src/Schemes/AffineSchemes/Objects/Constructors.jl

```diff
@@ -0,0 +1,670 @@
+
+julia> I = ideal(R, [x]);
+
+julia> U = complement_of_ideal(ideal(R, [y]));
+
+julia> Spec(R, I, U)
+Spec of Localization of Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal(x) at the multiplicative set complement of ideal(y)
+```
+"""
+Spec(R::MPolyRing, I::MPolyIdeal, U::AbsMPolyMultSet) = Spec(MPolyQuoLocalizedRing(R, I, U))
+
+
+
+########################################################
+# (2) Copy constructors
+########################################################
+
+@doc Markdown.doc"""
+    Spec(X::Spec)
+
+For convenience, an affine spectrum can be passed to ``Spec`
+to create a new spectrum. This can be particularly useful
+when in need to copy an affine spectrum.
+
+# Examples
+```jldoctest
+julia> R, (x, y) = PolynomialRing(QQ, ["x", "y"]);
+
+julia> I = ideal(R, [x]);
+
+julia> X = Spec(R, I)
+Spec of Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal(x)
+
+julia> Y = Spec(X)
+Spec of Quotient of Multivariate Polynomial Ring in x, y over Rational Field by ideal(x)
+```
+"""
+Spec(X::Spec) = Spec(OO(X))
+Base.deepcopy_internal(X::Spec, dict::IdDict) = Spec(deepcopy_internal(OO(X), dict))
+
+
+
+########################################################
+# (3) Affine n-dimensional space
+########################################################
+
+@doc Markdown.doc"""
+    affine_space(kk::BRT, n::Int; variable_name="x") where {BRT<:Ring}
+
+The ``n`-dimensional affine space over a ring ``kk`` is created
+by this method. The variable names are chosen as ``x1, x2, \dots``
+by default. This choice can be overwritten with a third optional argument.
+
+# Examples
+```jldoctest
+julia> affine_space(QQ, 5)
+Spec of Multivariate Polynomial Ring in x1, x2, x3, x4, x5 over Rational Field
+
+julia> affine_space(QQ,5,variable_name="y")
+Spec of Multivariate Polynomial Ring in y1, y2, y3, y4, y5 over Rational Field
+```
+"""
+function affine_space(kk::BRT, n::Int; variable_name="x") where {BRT<:Ring}
+  R, _ = PolynomialRing(kk, [variable_name * "$i" for i in 1:n])
+  return Spec(R)
+end
+
+
+@doc Markdown.doc"""
+    affine_space(kk::BRT, var_symbols::Vector{Symbol}) where {BRT<:Ring}
+
+This method also creates the ``n`-dimensional affine space over
+a ring ``kk`, but allows more flexibility in the choice of variable
+names. The following example demonstrates this.
+
+# Examples
+```jldoctest
+julia> affine_space(QQ,[:y1,:z2,:a])
+Spec of Multivariate Polynomial Ring in y1, z2, a over Rational Field
+```
+"""
+function affine_space(kk::BRT, var_symbols::Vector{Symbol}) where {BRT<:Ring}
+  R, _ = PolynomialRing(kk, var_symbols)
+  return Spec(R)
+end
+
+
+########################################################
+# (4) StdSpec (needed?)
+########################################################
+
+@doc Markdown.doc"""
+    standard_spec(X::AbsSpec{<:Any, <:MPolyRing})
+
+For an affine spectrum whose coordinate ring is a polynomial
+ring, this method computes the standard spectrum.
```

Comment:
What is the standard spectrum? Is this just for internal reasons or is this user facing? In the latter case, the documentation needs to mention what it is and what it is good for.

Reply:
This is internal and does not necessarily need to be documented. It turns an arbitrary Spec into one with an `MPolyQuoLocalizedRing` inside.

Reply:
Then maybe just change the documentation to "this method ensures a standardized internal represenation of a Spec"

Reply:
It was also request by @simonbrandhorst above to remove the documentation for this method (at least for now). I have done so. The `doc_strings` are still there, but are not called in the documentation. So in case this is needed in the future, it will be easy to add this.